### PR TITLE
Move files tree in shared folder

### LINKED ISSUE #1172: File Management: User can "move" a file into the folder that it's already in
**Describe the bug**
User can "move" a file into the folder that it's already residing in

**To Reproduce**

1. Login to files
2. Create a folder in your drive root
3. Upload a file to your drive root
4. Move the uploaded file into recently created folder
5. Navigate to the new folder
6. Click the kebab menu icon for the file and select move
7. In the folder path selection, select the folder that the file is already residing in
8. Click the move button

**Observed behavior**
A "File moved successfully" toast will appear and the filename of the file will be appended with `(1)`

**Expected behavior**
It should not be possible to move a file into the folder it is already residing in

**Screenshot**
Screen Shot 2021-06-22 at 10 07 14 AM

**Desktop (please complete the following information):**

OS: macOS
Browser: Chrome
Version : 91

**Additional context**
Happening on dev and current production release

### LINKED ISSUE #1204: Move files inside shared browser isn't showing the right tree
**Describe the bug**
After #1048, When moving a file, the tree is not the right one (it's the one of not shared folders).

**To Reproduce**
Go to a shared folder and move a file.

### PR DESCRIPTION
**Pull request by tanmoyAtb**
closes #1204 
closes #1172 

I also added the disabling of path on same path.
I have changes from #1223 on this PR

### CODE REVIEW

**Review by FSM1 — APPROVED** [review]
working great :)

File: packages/files-ui/src/Components/Modules/FileBrowsers/MoveFileModal.tsx [hunk 1]
```diff
@@ -95,25 +94,40 @@ const MoveFileModule = ({ filesToMove, modalOpen, onClose, onCancel, mode }: IMo
...
-          }
-        ]
-        setFolderTree(folderTreeNodes)
+        if (bucket.type === "share") {
+          // share folder tree
+          const shareFolderTreeRoot = newFolderTree.entries[0].entries[0]
+          const folderTreeNodes = [
+            {
+              id: "/",
+              title: bucket.name || "Home",
```
Comment (Tbaut) [inline packages/files-ui/src/Components/Modules/FileBrowsers/MoveFileModal.tsx]: I think we should have something like the bucket id there, or "Root - {bucket id}". But this should be reflected also on the title etc, so prob not for this PR.

```suggestion
              title: bucket.name || "Root",
```

File: packages/files-ui/src/Components/Modules/FileBrowsers/MoveFileModal.tsx [hunk 2]
```diff
@@ -189,29 +203,44 @@ const MoveFileModule = ({ filesToMove, modalOpen, onClose, onCancel, mode }: IMo
...
-          className={classes.okButton}
-          loading={movingFile}
-          disabled={!movePath}
-          onClick={onMoveFile}
+        {movePath === currentPath && (
+          <Typography
+            component="p"
+            variant="body1"
+          >
+            <Trans>The files are already in this folder</Trans>
```
Comment (Tbaut) [inline packages/files-ui/src/Components/Modules/FileBrowsers/MoveFileModal.tsx]: We can get fancy here with the translation, we need to import `plural` from lingui.
```suggestion
            <Trans>
              {plural(filesToMove.length, {
                one: `The file is already in this folder`,
                other: `The files are already in this folder`
              })}
            </Trans>
```

File: packages/files-ui/src/Components/Modules/FileBrowsers/MoveFileModal.tsx [hunk 3]
```diff
@@ -95,25 +94,40 @@ const MoveFileModule = ({ filesToMove, modalOpen, onClose, onCancel, mode }: IMo
...
-            id: "/",
-            title: "Home",
-            isExpanded: true,
-            expandable: true,
-            tree: mapFolderTree(newFolderTree.entries)
-          }
-        ]
-        setFolderTree(folderTreeNodes)
+        if (bucket.type === "share") {
+          // share folder tree
```
Comment (Tbaut) [inline packages/files-ui/src/Components/Modules/FileBrowsers/MoveFileModal.tsx]:
```suggestion
          // shared folder tree
```

**Review by Tbaut — COMMENTED** [review]
Looking good, but found a bug that I'm not 100% sure it's from the api or from the PR:


https://user-images.githubusercontent.com/33178835/124454482-d4c67a00-dd88-11eb-992e-3f555f598148.mp4

**Review by Tbaut — APPROVED** [review]
The bug I found is not linked to this PR, so it's all good :rocket:

**Review by RyRy79261 — APPROVED** [review]

**Pull request merged**